src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/Internal/Deque.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
This looks like a dup of code from Channels.
https://github.com/dotnet/runtime/blob/d9afc1ecd6f22d836e1d7665847917aa09f580e9/src/libraries/System.Threading.Channels/src/System/Collections/Generic/Deque.cs
We should move that file to a shared location and use it in both places rather than duplicating it.